### PR TITLE
i#3303: add and fix ARM/AArch64 support for drreg-test6

### LINKED ISSUE #3303: ASSERT and CRASH on android drreg-test
Suspect culprit is PR #3301 since drreg-test passed every time before this:

http://dynamorio.org/CDash/testDetails.php?test=593927&build=39816
```
drreg-test running
<Application /data/local/tmp/build_android-debug-internal-32/suite/tests/bin/client.drreg-test (7782).  Internal Error: DynamoRIO debug check failure: /work/dr/nightly/src/core/dispatch.c:488 dr_get_isa_mode(dcontext) == FRAG_ISA_MODE(targetf->flags) IF_X64(|| (dr_get_isa_mode(dcontext) == DR_ISA_IA32 && !FRAG_IS_32(targetf->flags) && DYNAMO_OPTION(x86_to_x64)))
(Error occurred @3551 frags)
version 7.0.17873, custom build
-no_dynamic_options -client_lib '/data/local/tmp/build_android-debug-internal-32/suite/tests/bin/libclient.drreg-test.dll.so;0;' -code_api -stderr_mask 12 -stack_size 56K -signal_stack_size 32K -max_elide_jmp 0 -max_elide_call 0 -early_inject -emulate_brk -no_inline_ignored_syscalls -native_exec_default_list '' -no_nativ
0xb6d5b595 0x78af04b0
/data/local/tmp/build_android-debug-internal-32/suite/tests/bin/libclient.drreg-test.dll.so=0xb6c97000
/system/lib/libc.so=0xb6bed000
/system/lib/libdl.so=0xb6be8000
/data/local/tmp/build_android-debug-internal-32/ext/lib32/debug/libdrreg.so=0xb6c84000
/data/local/tmp/build_android-debug-internal-32/ext/lib32/debug/libdrmgr.so=0xb6c71000>
```

http://dynamorio.org/CDash/testDetails.php?test=593941&build=39817
```
drreg-test running
<Application /data/local/tmp/build_android-release-external-32/suite/tests/bin/client.drreg-test (8459).  Tool internal crash at PC 0xb6f67dfc.  Please report this at your tool's issue tracker.  Program aborted.
Received SIGSEGV at pc 0xb6f67dfc in thread 8459
Base: 0xb6ecc000
Registers:  r0 =0x00000000 r1 =0x00000000 r2 =0xb6fec000 r3 =0xeafffffe
	r4 =0x00000000 r5 =0x5154d610 r6 =0x00000000 r7 =0x00000000
	r8 =0x51545c88 r9 =0x51545c90 r10=0x00000000 r11=0x00000001
	r12=0x0000516a r13=0x51545c80 r14=0xb6f67d9d r15=0xb6f67dfc
	eflags=0x88071830
version 7.0.17873, custom build
-no_dynamic_options -client_lib '/data/local/tmp/build_android-release-external-32/suite/tests/bin/libclient.drreg-test.dll.so;0;' -code_api -stack_size 56K -signal_stack_size 32K -max_elide_jmp 0 -max_elide_call 0 -early_inject -emulate_brk -no_inline_ignored_syscalls -native_exec_default_list '' -no_native_exec_managed>
```

### PR DESCRIPTION
**Pull request by hgreving2304**
Add ARM/AArch64 support for 3rd DR spill slot test6 in drreg-test.

Fixes #3303

### CODE REVIEW

**Comment by hgreving2304** [comment]
I am currently not able to test this, in particular on Android. But since it is straight forward, I have added ARM support "blindly", it should work.

File: suite/tests/client-interface/drreg-test.c [hunk 1]
```diff
@@ -395,6 +395,14 @@ GLOBAL_LABEL(FUNCNAME:)
         add      REG_XSP, FRAME_PADDING /* make a legal SEH64 epilog */
         POP_CALLEE_SAVED_REGS()
         ret
+#elif defined(ARM)
+        /* XXX i#3289: prologue missing */
+        /* Test 6: doesn't exit for ARM */
```
Comment (derekbruening) [inline suite/tests/client-interface/drreg-test.c]: You mean "doesn't exist"??

File: suite/tests/client-interface/drreg-test.c [hunk 2]
```diff
@@ -395,6 +395,14 @@ GLOBAL_LABEL(FUNCNAME:)
...
+        bx       lr
+#elif defined(AARCH64)
+        /* XXX i#3289: prologue missing */
+        /* Test 6: doesn't exit for ARM */
```
Comment (derekbruening) [inline suite/tests/client-interface/drreg-test.c]: Ditto?

File: suite/tests/client-interface/drreg-test.c (hunk 1 above)
Comment (hgreving2304) [reply suite/tests/client-interface/drreg-test.c]: Thanks

File: suite/tests/client-interface/drreg-test.c (hunk 2 above)
Comment (hgreving2304) [reply suite/tests/client-interface/drreg-test.c]: ack

**Pull request merged**